docs/designs/elasticdl_operator.md

```diff
@@ -0,0 +1,276 @@
+  restartPolicy: Never
+  volumes:
+  - hostPath:
+      path: /data
+    name: elasticdl-test-mnist-master-volume-0
+```
+
+We could rewrite it as a custom ElasticDLJob object
+after the ElasticDL CRD is created.
+The following is a demo:
+
+```yaml
+apiVersion: "elasticdl.org/v1"
+kind: "ElasticAIJob"
+metadata:
+  name: "test-mnist"
+spec:
+  jobArgs:
+    modelZoo: /model_zoo
+    modelDef: mnist_functional_api.mnist_functional_api.custom_model
+    trainingData: /data/mnist/train
+    validationData: /data/mnist/val
+    output: /data/output
+    dataReaderParams: ""
+    minibatchSize: 64
+    numMinibatchesPerTask: 20
+    evaluationStep: 1000
+    checkpointStep: 1000
+    checkpointDir: /data/checkpoint
+    checkpointDirForInit: ""
+    keepCheckpointMax: 0
+  Master:
+    template:
+      spec:
+        restartPolicy: Never
+        priorityClassName: high
+        containers:
+        - name: test-mnist-master
+          image: elasticdl-mnist
+          imagePullPolicy: Never
+          resources:
+            limits:
+              cpu: 1
+              memory: 2048Mi
+            requests:
+              cpu: 0.2
+              memory: 1024Mi
+          volumeMounts:
+          - mountPath: /data
+            name: test-volume
+        volumes:
+        - name: test-volume
+          hostPath:
+            path: /data
+  PS:
+    replicas: 2
+    template:
+      spec:
+        restartPolicy: Never
+        priorityClassName: high
+        containers:
+        - name: test-mnist-ps
+          image: elasticdl-ps
+          imagePullPolicy: Never
+          resources:
+            limits:
+              cpu: 1
+              memory: 2048Mi
+            requests:
+              cpu: 0.2
+              memory: 1024Mi
+  Worker:
+    replicas: 10
+    template:
+      spec:
+        restartPolicy: Never
+        priorityClassName: low
+        containers:
+        - name: test-mnist-worker
+          image: elasticdl-mnist
+          imagePullPolicy: Never
+          resources:
+            limits:
+              cpu: 1
+              memory: 2048Mi
+            requests:
+              cpu: 0.2
+              memory: 1024Mi
+          volumeMounts:
+          - mountPath: /data
+            name: test-volume
+        volumes:
+        - name: test-volume
+          hostPath:
+            path: /data
+```
+
+## ElasticDL CRD
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: eaijobs.elasticdl.org
+spec:
+  group: elasticdl.org
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    served: true
+    storaged: true
+  names:
+    kind: ElasticAIJob
+    listKind: ElasticAIJobList
+    singular: elasticaiJob
+    plural: elasticdljobs
+    shortNames:
+    - eaijob
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          properties:
+            jobArgs:
+              properties:
+                modelZoo:
```

Comment:
Is there any way to define jobArgs as arbitrary key-value pairs? 
After we extend ElasticDL to support pytorch, xgboost, etc, there must be arguments change. We want to fix the CRD definition and support arbitrary key-value pairs as arguments.

Reply:
We could define jobArgs as an array, we use '^--([a-z0-9_]+)\s([a-z0-9_]+)$' to check the item.

Please refer to the example in the updated doc.